src/Calendar.js

```diff
@@ -18,390 +18,422 @@
-    const formatIndex = arguments[1];
-    const fields = {
-      y: 'year',
-      Y: 'year',
-      M: 'month',
-      d: 'day',
-      D: 'day',
-      h: 'hour',
-      H: 'hour',
-      m: 'minute',
-    };
-
-    const whichField = fields[(formatIndex < 3) ? this.dateFormat.split(/[^a-z]/i)[formatIndex].charAt(0) : this.timeFormatText.replace(/[a]\s/i, '').split(/[^a-z]/i)[formatIndex - 3].charAt(0)];
-    const whichFormat = whichTemplate === 'selectorTemplate' ? whichField : uCase(whichField);
 
-    return string.substitute(this[whichTemplate], [whichFormat]);
+    return this;
   },
-  show: function show(o) {
-    this.inherited(arguments);
-    const options = o || this.options;
-
-    this.titleText = options.label ? options.label : this.titleText;
-
-    this.showTimePicker = this.options && this.options.showTimePicker;
-
-    this.date = moment((this.options && this.options.date) || moment());
-
-    if (this._isTimeless()) {
-      this.date.add({
-        minutes: this.date.zone(),
-      });
+  changeMonthShown: function changeMonthShown({ month }) {
+    domConstruct.empty(this.monthNode);
+    this.monthNode.innerHTML = month;
+    return this;
+  },
+  changeYearShown: function changeYearShown({ year }) {
+    domConstruct.empty(this.yearNode);
+    this.yearNode.innerHTML = ' ' + year;
+    return this;
+  },
+  checkAndRenderDay: function checkAndRenderDay(data = {}) {
+    const dayIndexer = data.day + data.startingDay - 1;
+    if (data.day === data.todayMoment.date() && data.todayMoment.month() === data.dateMoment.month()) {
+      data.isToday = 'isToday';
+    } else {
+      data.isToday = '';
     }
-
-    this.year = this.date.year();
-    this.month = this.date.month();
-
-    const today = moment();
-
-    this.populateSelector(this.yearNode, this.year, (this.year < today.year() - 10) ? this.year : today.year() - 10, // min 10 years in past - arbitrary min
-      (10 + today.year()) // max 10 years into future - arbitrary limit
-    );
-
-    this.populateSelector(this.monthNode, this.month, 0, 11);
-    this.populateSelector(this.dayNode, this.date.date(), 1, this.daysInMonth());
-    this.populateSelector(this.hourNode,
-      this.date.hours() > 12 && !this.is24hrTimeFormat ? this.date.hours() - 12 : (this.date.hours() || 12),
-      this.is24hrTimeFormat ? 0 : 1,
-      this.is24hrTimeFormat ? 23 : 12
-    );
-
-    this.populateSelector(this.minuteNode, this.date.minutes(), 0, 59);
-
-    if (this.date.hours() < 12) {
-      domAttr.set(this.meridiemNode, 'toggled', true);
-      domClass.add(this.meridiemNode, 'toggleStateOn');
+    if (dayIndexer % 7 === data.weekEnds.Sunday || dayIndexer % 7 === data.weekEnds.Saturday) {
+      data.weekend = 'weekend';
     } else {
-      domAttr.set(this.meridiemNode, 'toggled', false);
-      domClass.remove(this.meridiemNode, 'toggleStateOn');
+      data.weekend = '';
     }
+    data.date = data.dateMoment.date(data.day).format('YYYY-MM-DD');
```

Comment:
data.dateMoment is mutated here. Did you mean to use `data.dataMoment.clone().date(...)`